lib/command/definitions.js

```diff
@@ -2,387 +2,160 @@ const getConfig = require('./utils').getConfig;
-  definitionsTemplate = definitionsTemplate.replace('{{exportPageObjects}}', exportPageObjects.join('\n'));
-  definitionsTemplate = definitionsTemplate.replace('{{injectPageObjects}}', injectPageObjects.join('\n'));
-  definitionsTemplate = definitionsTemplate.replace('{{callbackParams}}', `, ${[...callbackParams, '...args: any'].join(', ')}`);
-  if (translations) {
-    definitionsTemplate = definitionsTemplate.replace(/\{\{I\}\}/g, translations.I);
+    supportObject[name] = name;
   }
 
+  const definitionsTemplate = template({
+    helperNames,
+    supportObject,
+    importPaths: getImportString(testsPath, targetFolderPath, supportPaths, helperPaths),
+    translations,
+    hasCustomStepsFile,
+  });
+
   fs.writeFileSync(path.join(targetFolderPath, 'steps.d.ts'), definitionsTemplate);
   output.print('TypeScript Definitions provide autocompletion in Visual Studio Code and other IDEs');
   output.print('Definitions were generated in steps.d.ts');
 };
 
-function addAllMethodsInObject(supportObj, actions, methods, translations) {
-  for (const action of methodsOfObject(supportObj)) {
-    const fn = supportObj[action];
-    if (!fn.name) {
-      Object.defineProperty(fn, 'name', { value: action });
-    }
-    const actionAlias = translations ? translations.actionAliasFor(action) : action;
-    if (!actions[actionAlias]) {
-      methods.push(toTypeDef(fn));
-      actions[actionAlias] = 1;
-    }
-  }
-  return methods;
+/**
+ * Returns the relative path from the  to the targeted folder.
+ * @param {string} originalPath
+ * @param { string} targetFolderPath
+ * @param { string} testsPath
+ */
+function getPath(originalPath, targetFolderPath, testsPath) {
+  if (!originalPath.startsWith('.')) return originalPath;
+  const parsedPath = path.parse(originalPath);
+  return path.relative(targetFolderPath, path.join(testsPath, parsedPath.dir, parsedPath.name));
 }
 
-function addAllNamesInObject(supportObj, actions, methods) {
-  for (const name in supportObj) {
-    if (actions[name]) {
-      continue;
-    }
-    const actor = supportObj[name];
-    let params = toTypeDef(actor);
-    if (params !== undefined) {
-      if (params.indexOf(' : ') > 0) {
-        if (params.indexOf(' (') > 0) {
-          params = params.trim();
-          methods.push(`    ${(name)}${params}\n`);
-        } else {
-          methods.push(`${params}`);
-        }
-      } else {
-        methods.push(`    ${(name)}: (${params}) => any; \n`);
-      }
-    }
+/**
+ *
+ *
+ * @param {string} testsPath
+ * @param {string} targetFolderPath
+ * @param {Object<string, string>} pathsToType
+ * @param {Object<string, string>} pathsToValue
+ * @returns
+ */
+function getImportString(testsPath, targetFolderPath, pathsToType, pathsToValue) {
+  const importStrings = [];
+
+  for (const name in pathsToType) {
+    const relativePath = getPath(pathsToType[name], targetFolderPath, testsPath);
+    importStrings.push(`type ${name} = typeof import ('./${relativePath}');`);
```

Comment:
The path should not start with dot

![image](https://user-images.githubusercontent.com/11293201/62793015-3a485800-bad9-11e9-8ae1-a02c354b5f9a.png)